targets/csharp/templates/PlayFabUtil.cs.ejs

```diff
@@ -49,6 +51,100 @@ namespace PlayFab
                 return null;
             return error.GenerateErrorReport();
         }
+
+#if NET45 || NETCOREAPP2_0
+        [ThreadStatic]
+        private static StringBuilder _sb;
+        /// <summary>
+        /// A threadsafe way to block and load a text file
+        /// 
+        /// Load a text file, and return the file as text.
+        /// Used for small (usually json) files.
+        /// </summary>
+        public static string ReadAllFileText(string filename)
+        {
+            if (!File.Exists(filename))
+                return string.Empty;
+                
+            if (_sb == null)
+                _sb = new StringBuilder();
+            _sb.Length = 0;
+
+            FileStream fs = null;
+            try 
+            {
+                fs = new FileStream(filename, FileMode.Open);
+                using (BinaryReader br = new BinaryReader(fs))
+                {
+                    // Attempt to release the descriptor immediately
+                    fs = null;
+
+                    while (br.BaseStream.Position != br.BaseStream.Length)
+                        _sb.Append(br.ReadChar());
+                }
+            }
+            finally
+            {
+                // Make sure to dispose of the file stream in case the construction of binary reader fails
+                if (fs != null)
+                    fs.Dispose();
+            }
+            return _sb.ToString();
+        }
+
+        internal static string GetLocalSettingsFileProperty(string propertyKey)
+        {
+            string envFileContent = null;
+            string currDir = Directory.GetCurrentDirectory();
+            string currDirEnvFile = Path.Combine(currDir, _localSettingsFileName);
+
+            if (File.Exists(currDirEnvFile))
+            {
+                envFileContent = ReadAllFileText(currDirEnvFile);
+            }
+            else
+            {
+                string tempDir = Path.GetTempPath();
+                string tempDirEnvFile = Path.Combine(tempDir, _localSettingsFileName);
+
+                if (File.Exists(tempDirEnvFile))
+                {
+                    envFileContent = ReadAllFileText(tempDirEnvFile);
+                }
+            }
+            
+            if (!string.IsNullOrEmpty(envFileContent))
+            {
+                var jsonPlugin = PluginManager.GetPlugin<ISerializerPlugin>(PluginContract.PlayFab_Serializer);
+                dynamic envJson = jsonPlugin.DeserializeObject(envFileContent);
+                string result = string.Empty;
+                try
+                {
+                    result = envJson[propertyKey]?.ToString();
+                    return result;
+                }
+                catch (KeyNotFoundException)
+                {
+                    return string.Empty;
+                }
+            }
+            return string.Empty;
+        }
+
+        internal static List<string> GetLocalDebugFunctions()
```

Comment:
It feels odd having GetLocalSettingsFileProperty, which is general and GetLocalDebugFunctions which is specific. I'd expect withe GetLocalDebugUri/GetLocalDebugFunctions or GetLocalSettingsFileProperty/GetLocalSettingsFileArrayProperty.